.github/workflows/doc.yml

```
@@ -0,0 +1,41 @@
+name: docs
+on:
+  push:
+    branches:
+      - "main"
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    name: build and deploy documentation.
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v4
```

as per `makefile.yml`